modules/library/main/src/main/java/org/geotools/filter/function/color/HSLFunction.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2017, Open Source Geospatial Foundation (OSGeo)
+ *    
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.function.color;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import java.awt.Color;
+
+import org.geotools.filter.FunctionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+
+/**
+ * hsl lesscss.org color function.
+ *
+ * @author Andrea Aime - GeoSolutions
+ */
+public class HSLFunction extends FunctionImpl {
+
+    public static FunctionName NAME = new FunctionNameImpl("hsl", parameter("result", Color.class),
+            parameter("hue", Double.class), parameter("saturation", Double.class),
+            parameter("lightness", Double.class));
```

Comment:
hue is an angle, so between 0 and 360, saturation and lightness are percentages, between 0 and 1